### ISSUE TITLE
Lock all collector operations (including scanning) while a collector operation is in progress.

### ISSUE THREAD

**Issue by pmengelbert** [Azure/eraser#180]
**Describe the solution you'd like**
The collect-scan-erase cycle will run on a timer, and we want to avoid starting a new cycle while an existing one is in progress. This could be controlled by mutex, or by some sort of locking mechanism for the kubernetes resources themselves.

**Comment by kitarp29** [comment]
Hi @pmengelbert I have a lil experience with Golang.
I wanna get started with this issue.
Could you assist me with it?

**Comment by pmengelbert** [comment]
We have discussed this and decided this feature is not necessary. Instead, we should document that ImageJobs should not be run simultaneously and doing so may result in unexpected behavior.

We decided not to implement for two main reasons. First, the implementation was complicated, especially in releasing the lock (see comments on #236 ). Second, automated ImageJobs run on a 24 hour timer. It's very unlikely that manually triggered ImageJobs will conflict with the ones run on a timer. Such a situation can be considered an edge case.
